flexmeasures/data/schemas/sensors.py

```diff
@@ -98,12 +98,22 @@ class QuantityOrSensor(MarshmallowClickMixin, fields.Field):
     def __init__(
         self, to_unit: str, default_src_unit: str | None = None, *args, **kwargs
     ):
-        """
+        """Field for validating, serializing and deserializing a Quantity or a Sensor.
+
+        NB any validators passed are only applied to Quantities.
+        For example, validate=validate.Range(min=0) will raise a ValidationError in case of negative quantities,
+        but will let pass any sensor that has recorded negative values.
+
         :param to_unit: unit in which the sensor or quantity should be convertible to
         :param default_src_unit: what unit to use in case of getting a numeric value
         """
 
+        _validate = kwargs.pop("validate", None)
         super().__init__(*args, **kwargs)
+        if _validate is not None:
+            # Insert validation into self.validators so that multiple errors can be stored.
+            validator = RepurposeValidatorToIgnoreSensors(_validate)
```

Comment:
Nice!

Seeing this, it made me think that we should discuss how to validate Sensor data. 

Perhaps, this should be a timely_beliefs feature that, on UPSERT, it checks (optionally) whether the data is valid or not.

Reply:
There are at least three moments to consider:
1. When sensor data is UPSERTed
2. When a trigger references a sensor in its flex-model or flex-context
3. When a scheduler GETs sensor data

Gathering some thoughts:

### 1. When sensor data is UPSERTed

- It's still unknown whether the sensor will even be referenced in a flex-model or flex-context.

### 2. When a trigger references a sensor

- Allows early validation whose results can be immediately returned to the user.
- Sensor data may still get updated in the time between triggering (creating the scheduling job) and scheduling (executing the scheduling job).

### 2. When a scheduler GETs sensor data

- Validation against the exact data that would be used in scheduling
- Validation errors would show up in failed task logs.

Reply:
Good ideas!

I've converted this to a discussion: https://github.com/FlexMeasures/flexmeasures/discussions/994?converting=1.